components/Main/index.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react'
+import { View, Text, TextInput, TouchableOpacity, ScrollView } from 'react-native'
+import Item from 'components/Item'
+import './index.styl'
+import { observer } from 'startupjs'
+
+export default observer(function App () {
+
+  const [input, setInput] = useState('')
+  const[item, setItem] = useState([])
+
+  const AddTodo = () => {
+    if(input) {
+      const date = new Date()
+      setItem([...item, {
+        text : input,
+        key : Date.now(),
+        completed : false,
+        date : date.getDay() +
+          "/" + (date.getMonth() + 1) +
+          "/" + date.getFullYear()
+      }])
+    }
+    setInput('')
+  }
+
+  const EditTodo = (id, newItem) => {
+    setItem(item.filter((edit) => {
+      if(edit.id === id) return edit.text = newItem
+      else return edit
+    }))
+  }
+
+  const DeleteTodo = (id) => {
+    setItem(item.filter((todo) => {
+      if (todo.key !== id) return true
+    }))
+  }
+
+  const TodoCompleted = (id) => {
+    setItem(item.filter((todo) => {
+      if (todo.key === id) todo.completed = !todo.completed
+      return todo
+    }))
+  }
+
+  const CompletedItems = (item) => item.filter(count => count.completed).length
```

Comment:
@TimmyByDeOverly when you will be using it won't be an appropriate way to calculate a count of todos.

Reply:
done